packages/phone-number-privacy/combiner/src/common/sign.ts

```diff
@@ -71,10 +71,19 @@ export abstract class SignAction<R extends OdisSignatureRequest> extends Combine
   }
 
   protected handleMissingSignatures(session: CryptoSession<R>) {
-    let error: ErrorType = ErrorMessage.NOT_ENOUGH_PARTIAL_SIGNATURES
+    let error: ErrorType
     const majorityErrorCode = session.getMajorityErrorCode()
-    if (majorityErrorCode === 403 || majorityErrorCode === 429) {
-      error = WarningMessage.EXCEEDED_QUOTA
+    switch (majorityErrorCode) {
+      case 403:
+      case 429:
+        error = WarningMessage.EXCEEDED_QUOTA
+        break
+      case 401:
+        // Authentication is checked in the combiner, but invalid nonces are passed through
```

Comment:
Feel free to add a TODO for this and we can get to it later

Reply:
Will add a TODO for this! Yeah we can always do something like have the abstract class call _handleCommon and _handleSpecific if there is a lot of overlap...
Was also wondering if it makes sense to define a different error code for nonce errors? To be fair, I couldn't think of what classical error would fit better, but it could be really nice for error handling to know the difference between signature issues + nonce issues?

Reply:
Yes (enough) since we also check signatures at the combiner level too (i.e. we duplicate those checks). That said, if there is a case where the combiner incorrectly forwards a request with bad auth, we can't really tell the difference right now

Reply:
Gotcha! I created this ticket to track https://github.com/orgs/celo-org/projects/164/views/28